lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -627,6 +628,19 @@ protected function prepareUpdateData($entity)
                 $fieldMapping = $this->class->fieldMappings[$field];
                 $columnName   = $fieldMapping['columnName'];
 
+                $isInsert = ! empty($this->queuedInserts[spl_object_id($entity)]);
```

Comment:
You can move this to line 609 already.

Reply:
Even better we can move a boolean flag nto `prepareUpdateData` for `isInsert` and directly pass it from `insert` or `update`.